audit.go

```diff
@@ -0,0 +1,162 @@
+package pagerduty
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/google/go-querystring/query"
+)
+
+const auditBaseURL = "/audit/records"
+
+// ListAuditRecordsOptions is the data structure used when calling the
+// ListAuditRecords API endpoint.
+type ListAuditRecordsOptions struct {
+	Actions              []string `url:"actions,omitempty,brackets"`
+	ActorID              string   `url:"actor_id,omitempty"`
+	ActorType            string   `url:"actor_type,omitempty"`
+	Cursor               string   `url:"cursor,omitempty"`
+	Limit                uint     `url:"limit,omitempty"`
+	MethodTruncatedToken string   `url:"method_truncated_token,omitempty"`
+	MethodType           string   `url:"method_type,omitempty"`
+	RootResourcesTypes   []string `url:"root_resources_types,omitempty,brackets"`
+	Since                string   `url:"since,omitempty"`
+	Until                string   `url:"until,omitempty"`
+}
+
+// ListAuditRecordsResponse is the response data received when calling the
+// ListAuditRecords API endpoint.
+type ListAuditRecordsResponse struct {
+	Records          []AuditRecord    `json:"records,omitempty"`
+	ResponseMetaData ResponseMetaData `json:"response_metadata,omitempty"`
+	Limit            uint             `json:"limit,omitempty"`
+	NextCursor       string           `json:"next_cursor,omitempty"`
+}
+
+// AuditRecord is a audit trail record that matches the query criteria.
+type AuditRecord struct {
+	ID               string           `json:"id,omitempty"`
+	Self             string           `json:"self,omitempty"`
+	ExecutionTime    string           `json:"execution_time,omitempty"`
+	ExecutionContext ExecutionContext `json:"execution_context,omitempty"`
+	Actors           []Actor          `json:"actors,omitempty"`
+	Method           Method           `json:"method,omitempty"`
+	RootResource     Resource         `json:"root_resource,omitempty"`
+	Action           string           `json:"action,omitempty"`
+	Details          Details          `json:"details,omitempty"`
+}
+
+// ResponseMetadata contains information about the response.
+type ResponseMetaData struct {
+	Messages []string `json:"messages,omitempty"`
+}
+
+// ExecutionContext contains information about the action execution context.
+type ExecutionContext struct {
+	RequestID     string `json:"request_id,omitempty"`
+	RemoteAddress string `json:"remote_address,omitempty"`
+}
+
+// Actor is the actor who carried out the action.
+type Actor APIObject
```

Comment:
I think we should avoid this becoming a named type. If PagerDuty ever adds any functionality in the future, where something else is referred to as an "actor", we can no longer use this name without a breaking change. I think using it directly as an `APIObject` is more clear to consumers that it's not a hydrated object.